CONTRIBUTING.md

```diff
@@ -74,35 +74,63 @@ git pull upstream refs/pull/999/head
 # Pull Request Etiquette
 1. Keep pull requests small: < 500 maximum added lines of code with rare exceptions, < 200 lines preferred.
 2. 1 commit per pull request: this will make commit history in the main branch easier to follow. 
-- Making significant contributions in a single commit is hard.
-- To solve this, we can leverage the fact that we are operating in a fork of the main repository, and so rewriting our feature branch history has no chance of polluting the main repository's history.
-- Let's say we are in our local clone of our bystro repository fork, on our feature branch feature/999. This feature branch has 7 commits, and we wish to condense that into 1:
 
+
+If you have multiple commits in your feature branch already (the branch you are PR'ing), you can squash those changes:
   ```sh
   # Ensure we're on our local feature/999 branch.
   git checkout feature/999
   
-  # Ensure we have the latest data from our fork.
-  git pull origin feature/999
-  
-  # Keep all code changes, but roll back commit history 7 commits.
-  git reset --soft HEAD~7
+  # Bring our local branch up to date with our remote branch
+  # If the branch's remote is "origin" this will be equivalent to executing: git pull origin feature/999
+  git pull
+  ```
   
-  # Create a new commit, with the cumulative changes of the past 7 commits.
-  # This will overwrite your local feature/999 branch's history, such that the last 7 commit records are replaced with a single commit that has the cumulative changes of those last 7 commits.
+  To squash all commits in the feature branch:
+  ```sh
+  # Squashes all commits in the feature branch
+  # Here we are assuming the bystrogenomics repository is the "upstream" remote
+  # e.g., that you ran `git remote add upstream git@github.com:bystrogenomics/bystro.git`
+  git reset --soft upstream/master
+  ```
+
+  Then, give your squashed commits a message, and push to remote:
+  ```sh
+  # Give your squashed commit a message
   git commit -a -m "Fixes performance regression, addressing issue #100"
   
-  # Double check that remote origin points to "foobar/bystro.git"
-  git remote --verbose
-  
   # Overwrite your fork's remote history to mirror the local history (with the past 7 commits condensed into 1).
   # With --force-with-lease, the push will be rejected if there are new commits on the remote branch that you have not pulled.
-  git push --force-with-lease origin feature/999
+  git push --force-with-lease
   ```
 
- - This will ensure that git history remains clean, while still allowing you to commit frequently during development.
+  If you have already created a branch from feature/999, which was just squashed, you can bring a child branch (feature/1000) up to date by:
```

Comment:
I think I would also add another subheading here, to indicate this section is just about child branches. I think the only part that really relates to etiquette is really lines 75 and 76, so maybe a subheading to indicate the different procedures would be useful. Like if line 79 had a subheading with something like 'How to maintain 1 commit per PR' for instance.